runtime-feature-detection.md

```diff
@@ -0,0 +1,156 @@
+# Runtime Feature Detection on Graviton
+
+Different generations of Graviton support different features. For example, Graviton3 supports SVE but Graviton2 does
+not. Graviton4 supports SVE2 and SVE. For some applications it may be adventageous to implement performance critical
+kernels to make use of the highest performing feature set available which may not be known until runtime. For this, the
+best practice is to consult HWCAPS. This guide covers the procedure for feature detection on Linux.
+
+On Linux, the standard C library implements `getauxval` to read from the auxiliary vector. Reading the `AT_HWCAP` and
+`AT_HWCAP2` values will return the hardware capability bitmaps which are filtered by the kernel to include only features
+which the kernel also supports. The library function is defined in `sys/auxv.h` and the constants for masking the bitmap
+are defined in `asm/hwcap.h`. Below is a simple C program to print all of the hardware capabilities and print an
+asterisk next to features which are supported by your CPU and Kernel combination.
+
+You will need to compile files which contain instructions or intrinsics for specific features with additional compiler
```

Comment:
This paragraph didn't follow for me.  Initially you are telling me about features, and how to detect them, but now it sounds like in order to detect features I need to enable SVE?

I think what you are showing is how to write a feature-specific path.  Maybe a more self-contained example:

```c
#include <stdio.h>
#include <sys/auxv.h>
#include <asm/hwcap.h>

def foo_sve() {
    .arch armv8-2+sve { ... }
    /* or */
#pragma GCC target("+sve")
}

def foo() {
    int have_sve = !!(getauxval(AT_HWCAP) & HWCAP_SVE);
    if (have_sve) {
        foo_sve();
    } else {
        foo_scalar();
    }
}
```

but actually won't this fail to compile on non-sve system? or does the arch/pragma skip it at compile time?

Reply:
actually, I assume it's worth pointing out that you shouldn't call getauxval everytime, as that's likely expensive....

Reply:
I proposed a new sample function, more like this. I forgot to mention that you should call getauxval every time, though.